### PR TITLE
boards: x86: acrn : Add configurations for acrn_ehl_crb

### PR DESCRIPTION
**Pull request by aasthagr**
Adding acrn configurations specific to the platform
on which acrn boots zephyr, Only the EHL specifc
configurations for now. Keeping the HW clock frequency to
1900Mhz for EHL and using the new APIC timer driver.

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

### CODE REVIEW

File: boards/x86/acrn/Kconfig.defconfig [hunk 1]
```diff
@@ -2,5 +2,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD
-	default "acrn"
+	default "acrn_ehl_crb"
```
Comment (nashif) [inline boards/x86/acrn/Kconfig.defconfig:5]: why is this needed?

File: boards/x86/acrn/acrn_ehl_crb.dts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
```
Comment (nashif) [inline boards/x86/acrn/acrn_ehl_crb.dts:7]: you can just `#include "acrn.dts"` here.

File: tests/benchmarks/boot_time/testcase.yaml [hunk 3]
```diff
@@ -2,6 +2,6 @@ tests:
   benchmark.kernel.boot_time:
     arch_allow: x86 arm posix
     platform_exclude: qemu_x86 qemu_x86_coverage qemu_x86_64 qemu_x86_nommu
-      minnowboard acrn
+      minnowboard acrn_ehl_crb
```
Comment (nashif) [inline tests/benchmarks/boot_time/testcase.yaml:5]: you should leave acrn here

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
> removes the generic acrn configurations and adding
> it specific to the platform on which acrn boots zephyr,
> only the EHL specifc configurations for now.
> Changing the HW clock frequency to 1900Mhz for EHL and
> using the new APIc timer driver.

You are not removing anything, acrn is still there. Do you really want to remove it? Dont we have acrn used on other platforms?

**Comment by aasthagr** [comment]
> > removes the generic acrn configurations and adding
> > it specific to the platform on which acrn boots zephyr,
> > only the EHL specifc configurations for now.
> > Changing the HW clock frequency to 1900Mhz for EHL and
> > using the new APIc timer driver.
> 
> You are not removing anything, acrn is still there. Do you really want to remove it? Dont we have acrn used on other platforms?

Commited the changes to remove acrn. If acrn is run on different platforms then the idea is to have separate .dts, .defconfig , .yaml files. I thought to make it similar to qemu_x86_mmu. Let me know if this is not what we want exactly?

**Comment by jhedberg** [comment]
@aasthagr wouldn't it make more sense to do a variant of `ehl_crb` rather than extend the `acrn` board (which I thought we wanted to remove eventually)? This would be analogous to how we have a `up_squared_32` board variant for `up_squared` (both of which can be found in `boards/x86/up_squared`)

**Comment by nashif** [comment]
> Commited the changes to remove acrn.

and what happens if someone has been using this on Apollo Lake or some other platform? If we are to make this HW specific then I would change the name completely and remove potential confusion. the other option is to do what @jhedberg is proposing. My concern is that right now this is a very basic and featureless configuration, once you start adding the building blocks to make it support use-cases, you might end up with more commonality with what the underlying hardware supports and needs.

**Comment by aasthagr** [comment]
> and what happens if someone has been using this on Apollo Lake or some other platform? If we are to make this HW specific then I would change the name completely and remove potential confusion. the other option is to do what @jhedberg is proposing. My concern is that right now this is a very basic and featureless configuration, once you start adding the building blocks to make it support use-cases, you might end up with more commonality with what the underlying hardware supports and needs.

@nashif What name do you suggest to remove the confusion? In this PR I have it as it is acrn_ehl_crb. I can add the generic configurations back (acrn.yaml, acrn_defconfig, acrn.dts) along with the platform specific configurations like acn_ehl_crb (or whatever name you suggest) if we think people might might have been using it on other platforms for now. Let me know your thoughts and any alternative suggestions? 

@jhedberg If we have the variant of already existing boards like ehl_crb, then as Anas suggested going forward we might have more commonality among the configurations for acrn rather than the existing hardware platforms. In that case, would it not be better to extend acrn board rather?

**Comment by jhedberg** [comment]
@aasthagr since acrn is a hypervisor and not a vm, I'd expect the available hardware that's presented to the guest OS to be very close to the real hardware. As such, deriving the board used from the "real HW" board definition seems like the most natural way forward to me. E.g. how else are the various EHL-specific devices in `dts/x86/elkhart_lake.dts` available for Zephyr if the acrn (or acrd-derived) board definition is used?

**Comment by aasthagr** [comment]
> @aasthagr since acrn is a hypervisor and not a vm, I'd expect the available hardware that's presented to the guest OS to be very close to the real hardware. As such, deriving the board used from the "real HW" board definition seems like the most natural way forward to me. E.g. how else are the various EHL-specific devices in `dts/x86/elkhart_lake.dtsi` available for Zephyr if the acrn (or acrd-derived) board definition is used?

@jhedberg ACRN hypervisor emulates all devices for ACRN VM(Zephyr). Therefore all devices(including uarts) in ACRN VM are virtualized or passthru. When we build on acrn as the board then ia32.dtsi is used instead of elkhart_lake.dtsi.  This is how the configurations are different from the actual board. At this point from my understanding, its a matter of deciding on where to keep the new board definitions. Let me know if we still think its a better idea to place these acrn_ehl_crb definitions under ehl_crb.

**Review by nashif — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by jhedberg** [comment]
@dcpleung @nashif was there some offline discussion that I missed?
